File: app/src/main/java/com/wenitech/cashdaily/Adapter/RecyclerViewGastoAdapter.java
package com.wenitech.cashdaily.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.wenitech.cashdaily.common.pojo.Gasto;
import com.wenitech.cashdaily.R;

import java.text.SimpleDateFormat;
import java.util.Date;

public class RecyclerViewGastoAdapter extends FirestoreRecyclerAdapter<Gasto, RecyclerViewGastoAdapter.mViewHolder> {

    public RecyclerViewGastoAdapter(@NonNull FirestoreRecyclerOptions<Gasto> options) {
        super(options);
    }


    @Override
    protected void onBindViewHolder(@NonNull final mViewHolder holder, int position, @NonNull Gasto model) {

        holder.valor.setText(String.valueOf(model.getValor()));
        holder.descripcion.setText(model.getDescripcion());

        SimpleDateFormat dateFormat = new SimpleDateFormat("yyy-MM-dd");
        Date date = model.getFecha().toDate();
        holder.fecha.setText(dateFormat.format(date));

        holder.cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
    }

    @NonNull
    @Override
    public mViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_gasto,parent,false);
        return new mViewHolder(view);
    }

    public class mViewHolder extends RecyclerView.ViewHolder{

        private CardView cardView;
        private Context context;
        private TextView valor;
        private TextView descripcion;
        private TextView fecha;

        public mViewHolder(@NonNull View itemView) {
            super(itemView);

            cardView = itemView.findViewById(R.id.cardViewItemGastoId);
            context = itemView.getContext();

            valor = itemView.findViewById(R.id.tv_valor);
            descripcion = itemView.findViewById(R.id.tv_descri);
            fecha = itemView.findViewById(R.id.tv_fecha);


        }
    }
}

File: app/src/main/java/com/wenitech/cashdaily/framework/features/caja/ui/adapter/RecyclerViewGastoAdapter.kt
package com.wenitech.cashdaily.framework.features.caja.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.wenitech.cashdaily.R
import com.wenitech.cashdaily.framework.commons.BaseViewHolder
import com.wenitech.cashdaily.domain.entities.CashTransactions
import com.wenitech.cashdaily.databinding.ItemMovimientoCajaBinding
import java.text.NumberFormat
import java.text.SimpleDateFormat

class RecyclerViewGastoAdapter(
    private val listener: Listener
) : RecyclerView.Adapter<BaseViewHolder<*>>() {

    interface Listener {
        fun onClickItemMovementBox(item: CashTransactions, position: Int)
    }

    private var data: List<CashTransactions> = listOf()

    fun setData(data: List<CashTransactions>) {
        this.data = data
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<*> {
        val viewBinding =
            ItemMovimientoCajaBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return mViewHolder(viewBinding)
    }

    override fun onBindViewHolder(holder: BaseViewHolder<*>, position: Int) {
        when (holder) {
            is mViewHolder -> holder.bind(data[position], position)
        }
    }

    private inner class mViewHolder(val binding: ItemMovimientoCajaBinding) :
        BaseViewHolder<CashTransactions>(binding.root) {

        override fun bind(item: CashTransactions, position: Int) {

            val stringValor: String
            val stringDate: String
            val numberFormatMoney = NumberFormat.getCurrencyInstance()
            stringValor = numberFormatMoney.format(item.value)
            val dateFormat = SimpleDateFormat("dd MMMM, yyyy")
            val date = item.serverTimestamp!!.toDate()
            stringDate = dateFormat.format(date)

            binding.tvValor.text = stringValor
            binding.tvDate.text = stringDate
            binding.tvDescription.text = item.description
            if (item.isExpense) {
                binding.imageViewPerfil2.setImageResource(R.drawable.ic_trending_up)
            } else {
                binding.imageViewPerfil2.setImageResource(R.drawable.ic_trending_down)
            }

            // On Click Listener
            binding.root.setOnClickListener {
                listener.onClickItemMovementBox(item, position)
            }


        }
    }

    override fun getItemCount(): Int {
        return data.size
    }

}